gcc unused parameter warnings with -Wextra

Hi, thanks for the product. I use '-pedantic -Wall -Wextra'. The '-Wextra' is causing these warnings:

E:/D_1/w rk/g_d/gezz/xinc/thrust-1.8.2/thrust/detail/allocator/allocator_traits.inl:181:27: warning: unused parameter 'a' [-Wunused-parameter]
     max_size(const Alloc &a)
     
E:/D_1/w rk/g_d/gezz/xinc/thrust-1.8.2/thrust/device_malloc_allocator.h:143:49: warning: unused parameter 'cnt' [-Wunused-parameter]
     inline void deallocate(pointer p, size_type cnt)

E:/D_1/w rk/g_d/gezz/xinc/thrust-1.8.2/thrust/detail/allocator/allocator_traits.inl:205:19: warning: unused parameter 'a' [-Wunused-parameter]
     system(Alloc &a)

--Thanks again.

--- a/thrust/detail/allocator/allocator_traits.inl
+++ b/thrust/detail/allocator/allocator_traits.inl
@@ -178,7 +178,7 @@ __host__ __device__
     has_member_max_size<Alloc>::value,
     typename allocator_traits<Alloc>::size_type
   >::type
-    max_size(const Alloc &a)
+    max_size(const Alloc &)
 {
   typedef typename allocator_traits<Alloc>::size_type size_type;
   return thrust::detail::integer_traits<size_type>::const_max;
@@ -202,7 +202,7 @@ __host__ __device__
     has_member_system<Alloc>::value,
     typename allocator_system<Alloc>::type
   >::type
-    system(Alloc &a)
+    system(Alloc &)
 {
   // return a copy of a default-constructed system
   typename allocator_system<Alloc>::type result;

--- a/thrust/device_malloc_allocator.h
+++ b/thrust/device_malloc_allocator.h
@@ -142,6 +142,9 @@ template<typename T>
     __host__
     inline void deallocate(pointer p, size_type cnt)
     {
+      // silence unused parameter warning while still leaving the parameter name for Doxygen
+      (void)(cnt);
+
       device_free(p);
     } // end deallocate()
 

